src/nss/new_nss_oslogin.c

```diff
@@ -72,6 +73,26 @@
        __typeof__ (b) _b = (b); \
      _a < _b ? _a : _b; })
 
+// Locking implementation: use pthreads.
+static pthread_mutex_t pwmutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_mutex_t grmutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Should these mutexes have attribute PTHREAD_MUTEX_ERRORCHECK or PTHREAD_MUTEX_NORMAL? The default initialization has "undefined" behavior for recursive locking.